firmware-support/bittide-hal-c/include/bittide_gather.h

```diff
@@ -7,73 +7,47 @@
 
 #include <stdint.h>
 
-/**
- * GatherUnit - Handle for interfacing with a gather unit
- *
- * The gather unit collects data from local memory and sends it to other nodes
- * over the Bittide network. The gather memory is double-buffered and swaps
- * buffers at metacycle boundaries according to the calendar configuration.
- */
-typedef struct {
-    volatile uint64_t* gather_memory;       ///< Pointer to gather memory base address
-    volatile uint32_t* metacycle_count;     ///< Pointer to metacycle count register
-    volatile uint32_t* metacycle_register;  ///< Pointer to metacycle synchronization register
-    uint32_t memory_len;                    ///< Length of gather memory in uint64_t words
-} GatherUnit;
-
-/**
- * Initialize a GatherUnit
- *
- * @param gather_memory Pointer to the gather memory base address
- * @param metacycle_count Pointer to the metacycle count register
- * @param metacycle_register Pointer to the metacycle synchronization register
- * @param memory_len Length of the gather memory in uint64_t words
- * @return Initialized GatherUnit struct
- */
-GatherUnit gather_unit_init(
-    volatile uint64_t* gather_memory,
-    volatile uint32_t* metacycle_count,
-    volatile uint32_t* metacycle_register,
-    uint32_t memory_len
-);
+#ifdef HAL_SCATTER_GATHER_PE_DEVICE_GATHER_UNIT_H
```

Comment:
Maybe a comment that explains how this file is supposed to be used would be nice?